x-pack/plugins/cases/common/types/api/case/v1.ts

```diff
@@ -135,6 +135,27 @@ export const CasePostRequestRt = rt.intersection([
   ),
 ]);
 
+/**
+ * Bulk create cases
+ */
+
+const CaseCreateRequestWithOptionalId = rt.intersection([
+  CasePostRequestRt,
+  rt.exact(rt.partial({ id: rt.string })),
+]);
+
+export const BulkCreateCasesRequestRt = rt.strict({
+  cases: rt.array(CaseCreateRequestWithOptionalId),
```

Comment:
I really like the API contract as by default I would have thought about just an array of cases.